### PR TITLE
Windows: Note about volumes in run

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>

@thaJeztah FYI. Adding on account of this comment: https://github.com/Microsoft/Virtualization-Documentation/issues/497#issuecomment-262636687

### CODE REVIEW

File: docs/reference/commandline/run.md [hunk 1]
```diff
@@ -262,6 +262,13 @@ https://docs.docker.com/engine/installation/binaries/#/get-the-linux-binary)),
 you give the container the full access to create and manipulate the host's
 Docker daemon.
 
+On Windows, the paths must be specified using Windows-style semantics. Windows
+containers do not support the target of a volume or bind-mount being pre-
+populated in the container image.
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md]: > Windows containers do not support the target of a volume or bind-mount being pre-populated in the container image.

Not sure I understand this part; Linux does not support this either? It's either

- `-v c:\target`, anonymous volume
- `-v namedvolume:c:\target`, named volume
- `-v c:\foo:c:\target`, bind-mount host directory (**not** a volume btw)

Comment (thaJeztah) [inline docs/reference/commandline/run.md]: oh, wait, I see; "pre-populated" as in, contain the data from the host.

Is that going to be supported? What is the use of a bind-mounted directory in that case, that's a big difference in semantics

Comment (jhowardmsft) [inline docs/reference/commandline/run.md]: It c:\target exists in the image, you will get back an error such as the following (the directory is not empty bit specifically)

```
PS C:\> docker run -v "c:\foo:c:\Program Files" microsoft/nanoserver cmd /s /c type "c:\Program Files\somefile.txt"
e:\go\src\github.com\docker\docker\bundles\docker.exe: Error response from daemon: container c8ac5e4ece96027744088b4a19694e5a9e1596bd9f5f16055018e2d38f8d215b encountered an error during CreateContainer: failure in a Windows system call: The directory is not empty. (0x91) extra info: {"SystemType":"Container","Name":"c8ac5e4ece96027744088b4a19694e5a9e1596bd9f5f16055018e2d38f8d215b","Owner":"docker","IsDummy":false,"VolumePath":"\\\\?\\Volume{a12acb49-a5f4-11e6-a9e3-00155d768276}","IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\control\\windowsfilter\\c8ac5e4ece96027744088b4a19694e5a9e1596bd9f5f16055018e2d38f8d215b","Layers":[{"ID":"fefb26f1-cf06-5de2-979b-54e24b4df8e7","Path":"C:\\control\\windowsfilter\\3ad1811989c1c273f1216de7ff6e36cc4374e031bead3fb32aaddcb19386ac6b"}],"HostName":"c8ac5e4ece96","MappedDirectories":[{"HostPath":"c:\\foo","ContainerPath":"c:\\program files","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0}],"HvPartition":false,"EndpointList":["55eedba6-5d07-47f5-b1ee-bc09e4609629"],"Servicing":false,"AllowUnqualifiedDNSQuery":true}.
PS C:\>
```

Comment (jhowardmsft) [inline docs/reference/commandline/run.md]: As opposed to Linux, where you can splat existing contents:

image

Comment (jhowardmsft) [inline docs/reference/commandline/run.md]: > Is that going to be supported?

We know it's a deficiency, has been forever (well TP4 when the functionality was introduced), but fixing it is hard by design of the mapped directories in the platform. 

> What is the use of a bind-mounted directory in that case, that's a big difference in semantics

You can still use bind-mounts just fine if the target doesn't exist. That's the caveat.

Comment (jhowardmsft) [inline docs/reference/commandline/run.md]: Oh hang on, I think you're mis-understanding. I'm saying if the target inside the container exists. Not on the host.

Comment (jhowardmsft) [inline docs/reference/commandline/run.md]: Exactly

Comment (johnstep) [inline docs/reference/commandline/run.md]: Makes sense; maybe add an example to clarify the pre-populated (target exists) case?

**Comment by jhowardmsft** [comment]
OK, made this a lot clearer and with examples.

File: docs/reference/commandline/run.md [hunk 2]
```diff
@@ -262,6 +262,26 @@ https://docs.docker.com/engine/installation/binaries/#/get-the-linux-binary)),
...
+	
+    PS C:\> docker run -v c:\foo:d: microsoft/nanoserver cmd /s /c type d:\somefile.txt
+    Contents of file
+
+The following examples will fail when using Windows-based containers, as the 
+destination of a volume or bind-mount inside the container must be one of: 
+a non-existing or empty directory; or a drive other than C:. Further, the source
+of a bind mount must be a local directory, not a file.
+
+    net use c: \\remotemachine\share
```
Comment (johnstep) [inline docs/reference/commandline/run.md]: Should this be `net use z: ...`?

Comment (jhowardmsft) [inline docs/reference/commandline/run.md]: Ah yes, thanks. Updated.

**Comment by PatrickLang** [comment]
Thanks! That's definitely clearer

**Review by johnstep — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
thanks, definitely better!

LGTM

**Pull request merged**